### ISSUE TITLE
The security token included in the request is invalid 

### ISSUE THREAD

**Issue by asyba** [aws/aws-toolkit-jetbrains#2020]
**Describe the bug**
After upgrading SAM CLI from 0.XX version to 1.1.0
I can't make it work.
I verified and the profile is working fine through a terminal with AWS CLI.
I tried in EAP version and normal version.

26 Aug 2020 13:08:58,214 [INFO] (lambdainternal.AWSLambda.startRuntime(AWSLambda.java:356)) Invoke failed with Throwable (Exception: java.lang.Throwable, Fatal: false )
The security token included in the request is invalid (Service: AWSSecretsManager; Status Code: 400; Error Code: UnrecognizedClientException; Request ID: d9bf95d9-5f61-4b20-907d-2f4bb46ba3a0): .......the security token included in the request is invalid (Service: AWSSecretsManager; Status Code: 400; Error Code: UnrecognizedClientException; 

**Your Environment**
- OS: Ubuntu 18.04.5 LTS
- JetBrains product: IntelliJ IDEA 2020.2.1 (Ultimate Edition)
- JetBrains product version: Build #IU-202.6948.69, built on August 24, 2020
- AWS Toolkit version: 1.17-202
- SAM CLI version: 1.1.0
- JVM/Python version: Java8

**Comment by asyba** [comment]
More Info the invalid token error I got it when Debugging.
Doing a normal execution I got: Exception in thread "main" java.lang.Error: java.lang.OutOfMemoryError: Metaspace

**Comment by asyba** [comment]
![image](https://user-images.githubusercontent.com/1466530/91309127-9c2c3c00-e786-11ea-881e-dbe6c4cf8d42.png)

**Comment by asyba** [comment]
could be related to this https://github.com/aws/aws-sam-cli/issues/2143 ?

**Comment by abrooksv** [comment]
Yep, that looks to be related.

**Comment by asyba** [comment]
> Yep, that looks to be related.

do you know how I can downgrade or install another older version of aws sam cli with brew?

**Comment by abrooksv** [comment]
Sorry, I never remember how to with brew.

We traced the issue, it happens with the Java8 runtime and using static access key / secret key, instead of access/secret/session token credentials.

The java11 runtime does not have the same issue nor the new java8.al2 (which the toolkit does not support yet)

**Comment by asyba** [comment]
with java11 manage to debug and it works.
I will close the issue, sam bug related.

**Comment by leo731121** [comment]
> > Yep, that looks to be related.
> 
> do you know how I can downgrade or install another older version of aws sam cli with brew?

hi, asyba,

This is the script to downgrade the sam local version.

git clone https://github.com/awslabs/aws-sam-cli
cd aws-sam-cli
git checkout tags/v0.53.0
python3 setup.py install
sam --version

**Comment by mateusarruda** [comment]
I'm having the same problem with version 1.15.0 on a Windows 10 OS.  After debugging I found out that the in the AmazonHttpClient.java in getCredentialsFromContext method the return is a AWSCredentials with a sessionToken as an empty string (image below)
). Using another version of SAM CLI, this property doesn't even exists in the object while debugging.  The same thing happened if I'm using Ubuntu 18 or 20. 

![bpcfr](https://user-images.githubusercontent.com/77297192/105227475-22835900-5b40-11eb-9fa5-e1acefdee8ca.png)


To solve my problem I downgraded my SAM CLI version to 1.6.2, but I don't know why this is happening.

I think this bug should be reopened and hope this information about the sessionToken to be helpful in any way.
